### DISCUSSION TITLE
Issue in lazy loading `torch` in surrogates

### DISCUSSION THREAD

**Discussion by rjavadi** [emdgroup/baybe#162] [Q&A] [answered]
I think importing `Tensor` for type checking causes `_structure_surrogate` to break. Test failure happens in `custom_pretrained.py` example with error: `cattrs.errors.StructureHandlerNotFoundError: Unsupported type: 'str'. Register a structure hook for it.`

**Comment by Scienfitz** [comment]
hmm I see a todo and workaround in that function, prob this is not a quick fix and def needs the sharp eyes of @AdrianSosic to look over

**Reply by rjavadi** [reply, marked as answer]
Found the reason and the solution:
https://github.com/python-attrs/cattrs/pull/42

**Comment by Scienfitz** [comment]
closing as answered
